java/src/main/native/include/jni_utils.hpp

```diff
@@ -721,7 +721,8 @@ inline void jni_cuda_check(JNIEnv *const env, cudaError_t cuda_status) {
     if (cudaErrorMemoryAllocation == cudaPeekAtLastError()) {                                      \
       cudaGetLastError();                                                                          \
     }                                                                                              \
-    JNI_CHECK_THROW_NEW(env, cudf::jni::OOM_CLASS, "Could not allocate native memory", ret_val);   \
+    auto what = std::string("Could not allocate native memory, caused by \n  ") + e.what();        \
```

Comment:
Nit: I'd rather see the exception detail on a single line here, makes it easier to grep out the details in a log.  Otherwise if one greps for `OutOfMemoryError` the details will be on the next line.
```suggestion
    auto what = std::string("Could not allocate native memory: ") + e.what();        \
```

Also, do we need to protect this code from `what()` returning `nullptr`?

Reply:
Thanks for review, updated.
Yes, better to have the protection.